### DISCUSSION TITLE
How to output [save_component_rate_in] in own app?

### DISCUSSION THREAD

**Discussion by Bearunner** [idaholab/moose#26007] [Q&A Modules: Porous Flow] [answered]
Dear all,

Is there any similar function, postprocessor, or auxkernel that can output the [save_component_rate_in] values in own app or MOOSE framework?

Thanks for your help.

 B

**Comment by Bearunner** [comment]
Dear @cpgr and @GiudGiud ,could you please help me solve this? Thanks a lot.

**Reply by GiudGiud** [reply]
Hello
What is this?
```
 [save_component_rate_in]
```
never seen it before. Are you talking about the `save_in` parameter?

**Reply by Bearunner** [reply]
It's shown in the Optional Parameters of [PorousFlowFullySaturated] action
https://mooseframework.inl.gov/source/actions/PorousFlowFullySaturated.html

C++ Type:std::vector<AuxVariableName>

Controllable:No

Description:List of AuxVariables into which the rate-of-change of each fluid component at each node will be saved. There must be exactly N of these to match the N fluid components. The result will be measured in kg/s, where the kg is the mass of the fluid component at the node (or m^3/s if multiply_by_density=false). Note that this saves the result from the MassTimeDerivative Kernels, but NOT from the MassVolumetricExpansion Kernels.

**Reply by GiudGiud** [reply, marked as answer]
ok I see it now

So this is custom code that is programmatically adding the right auxkernels with the right parameters to output these quantities.

You can absolutely do that in your own app, and use this code (or the MaterialPropertyOutput code) as an example.
You can also do this from the input file

**Reply by Bearunner** [reply]
Yes, Guillaume, thanks for your tips. 

But how can the 'rate-of-change of each fluid component at each node'  from the MassTimeDerivative Kernels be output?

**Reply by GiudGiud** [reply]
Ah you ll have to look in the code how they did it
I bet it s using the coupledDot (time derivative) on the fluid component (mass fraction? mass flow?)
